app/models/saved_job.rb

```diff
@@ -9,4 +9,30 @@ def self.new_by_job_key(job_key, user)
     job = Job.create_or_update_job_by_key(job_key)
     SavedJob.new(job: job, user: user)
   end
+
+  def self.check_for_current_user(jobs, user)
+    if user
+      has_current_user_saved_these_jobs(jobs, user)
+    else
+      no_saved_jobs(jobs)
+    end
+  end
+
+  def self.has_current_user_saved_these_jobs(jobs, user)
+    jobs.each do |job|
+      if user.saved_jobs.joins(:job).find_by("jobs.job_key": job[:job_key])
```

Comment:
Could we `#pluck` all job keys for a user and see if they are in this hash?  vs trigger a new DB call.